diff --git a/references_dialog.dialog_widgets.inc b/references_dialog.dialog_widgets.inc
index 4997e4b..afd0a5d 100644
--- a/references_dialog.dialog_widgets.inc
+++ b/references_dialog.dialog_widgets.inc
@@ -256,7 +256,7 @@ function references_dialog_entityreference_add_link($element, $widget_settings,
   // Create a link for each allowed bundles.
   if (isset($bundles)) {
     foreach ($bundles as $bundle) {
-      if (in_array($bundle, $entity_bundles) && $link = references_dialog_entityreference_link_helper($entity_type, $bundle)) {
+      if (in_array($bundle, $entity_bundles) && $link = references_dialog_entityreference_link_helper($entity_type, $bundle, $element)) {
         $add_links[] = $link;
       }
     }
@@ -264,7 +264,7 @@ function references_dialog_entityreference_add_link($element, $widget_settings,
   return $add_links;
 }
 
-function references_dialog_entityreference_link_helper($entity_type, $bundle = NULL) {
+function references_dialog_entityreference_link_helper($entity_type, $bundle = NULL, $element) {
   $wrapper = entity_metadata_wrapper($entity_type, NULL, array('bundle' => $bundle));
   $info = $wrapper->entityInfo();
   if (isset($bundle)) {
@@ -292,7 +292,7 @@ function references_dialog_entityreference_link_helper($entity_type, $bundle = N
       'title' => t('Create @type', array('@type' => $label)),
       'href' => $path,
     );
-    return $link;
+    return references_dialog_add_parent_query_param($link, $element);
   }
   return FALSE;
 }
